low-level/sensors/impl/generic/raid_integrity_data.py

```diff
@@ -170,22 +173,40 @@ def _raid_health_monitor(self):
                 # Check RAID device array state is 'idle' or not
                 result = self._check_raid_state(device)
                 if result == "failed":
+                    # Send alert
+                    self.alert_type = self.FAULT
+                    self._alert_msg = RaidAlertMsgs.STATE_MSG.value
+                    self._send_json_msg(self.alert_type, device, self._alert_msg)
+                    # Persist RAID device fault state
+                    self._fault_state = self.FAULT
+                    data = device + ":" + self._fault_state
+                    fault_status_file = self.DEFAULT_RAID_DATA_PATH + device + "_" +RaidDataConfig.RAID_SYNC_FAULT_STATUS.value
+                    self._create_file(fault_status_file)
+                    with open(fault_status_file, 'w') as fs:
+                        fs.write(data)
+                    # Retry to check RAID state
                     self._retry_execution(self._check_raid_state, device)
                 logger.debug("'idle' state is found in Raid device:{}."
                              .format(device))
 
                 # Check Mismatch count in RAID device files.
                 result = self._check_mismatch_count(device)
                 if result == "failed":
```

Comment:
Having just this check will still send alert again across system/SSPL resets. That can be handled at lower priority, I think.

Reply:
Done changes to handle such scenario